java/integTest/src/test/java/glide/ConnectionTests.java

```diff
@@ -67,6 +68,11 @@ public GlideClusterClient createAzTestClient(String az) {
                 .get();
     }
 
+    @SneakyThrows
+    private ValkeyCluster createDedicatedCluster(boolean clusterMode) {
```

Comment:
I propose to use `@BeforeAll` and `@AfterAll` to ensure that new cluster is started and stopped when it is needed. And ideally to move this test to another class, so it won't be started for other tests.
An alternative option is to start the cluster right inside the test and use `finally` block to stop it.